lib/metasploit/framework/login_scanner/redis.rb

```diff
@@ -0,0 +1,89 @@
+require 'metasploit/framework/login_scanner/base'
+require 'metasploit/framework/login_scanner/rex_socket'
+require 'metasploit/framework/tcp/client'
+
+module Metasploit
+  module Framework
+    module LoginScanner
+
+      # This is the LoginScanner class for dealing with REDIS.
+      # It is responsible for taking a single target, and a list of credentials
+      # and attempting them. It then saves the results.
+
+      class REDIS
+        include Metasploit::Framework::LoginScanner::Base
+        include Metasploit::Framework::LoginScanner::RexSocket
+        include Metasploit::Framework::Tcp::Client
+
+        DEFAULT_PORT         = 6379
+        LIKELY_PORTS         = [ DEFAULT_PORT ]
+        LIKELY_SERVICE_NAMES = [ 'redis' ]
+        PRIVATE_TYPES        = [ :password ]
+        REALM_KEY            = nil
+
+        # This method can create redis command which can be read by redis server
+        def redis_proto(command_parts)
```

Comment:
yeah, it is not possible to use Msf::Auxiliary::Redis for LoginScanner. You will have to do the other way around: Let Msf::Auxiliary::Redis use Metasploit::Framework::LoginScanner::REDIS. It's probably not a blocker for this PR though.